kroxylicious-filters/kroxylicious-simple-transform/src/main/java/io/kroxylicious/proxy/filter/simpletransform/Replacing.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.simpletransform;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.IllegalCharsetNameException;
+import java.nio.charset.StandardCharsets;
+import java.nio.charset.UnsupportedCharsetException;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.plugin.Plugin;
+import io.kroxylicious.proxy.plugin.PluginConfigurationException;
+
+@Plugin(configType = Replacing.Config.class)
```

Comment:
I added this then realised I could actually have achieved the same end result if I'd specified the charset for uppercasing in a secret. What do people think of keeping this vs reverting it?

Reply:
I think that String replacement lets us have more obviously "under-test" inputs. Changing a charset is a bit subtle compared to replacing with a string like "test-replacement". So I prefer this replacer from this debugging angle.

To integrate with secrets fully I think this should be in terms of `replaceValueFile` and the filter has to load it from disk. Otherwise I think we'll be replacing a string with the mounted filename.

Reply:
Haha you spotted the really tactical choice...

Ok I'll knock up a file based replacer.